src/components/LooMap/useLocateMapControl.ts

```diff
@@ -1,17 +1,28 @@
 import React from 'react';
-import L, { LatLngLiteral, Map } from 'leaflet';
+import L, { Map, LocationEvent, LayerGroup, LatLngExpression } from 'leaflet';
 import { fitMapBoundsToUserLocationNeighbouringTiles } from '../../lib/loo';
 
-const LocationMarker = L.Marker.extend({
-  initialize: function (latlng, options) {
+interface LocationMarkerOptions extends L.MarkerOptions {
+  color?: string;
+  fillColor?: string;
+  fillOpacity?: number;
+  opacity?: number;
+  weight?: number;
+  radius?: number;
+}
+
+class LocationMarker extends L.Marker {
+  options: LocationMarkerOptions;
+
+  constructor(latlng: LatLngExpression, options: LocationMarkerOptions) {
+    super(latlng, options);
     L.Util.setOptions(this, options);
-    // eslint-disable-next-line functional/immutable-data
-    this._latlng = latlng;
     this.createIcon();
-  },
+  }
 
-  _getIcon: function (options: { radius: number; weight: number }, style) {
-    const { radius, weight } = options;
+  private _getIcon(options: LocationMarkerOptions, style: string): L.DivIcon {
```

Comment:
To make it actually private, not just for TS, [you can prefix it with #](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties):

```suggestion
  #_getIcon(options: LocationMarkerOptions, style: string): L.DivIcon {
```